### PR TITLE
TT-3793 query batching config

### PR DESCRIPTION
**Pull request by urbanishimwe**
This PR adds functionalities to enable or disable graphql query batching from API definitions

## Related Issue
https://tyktech.atlassian.net/browse/TT-3793

## Motivation and Context
check description of the above related issue

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

### CODE REVIEW

File: apidef/api_definitions.go [hunk 1]
```diff
@@ -651,6 +651,8 @@ type GraphQLConfig struct {
 	Subgraph GraphQLSubgraphConfig `bson:"subgraph" json:"subgraph"`
 	// Supergraph holds the configuration for a GraphQL federation supergraph.
 	Supergraph GraphQLSupergraphConfig `bson:"supergraph" json:"supergraph"`
+	// QueryBatching is the query batching specific configuration.
+	QueryBatching GraphQLQueryBatching `bson:"query_batching" json:"query_batching"`
```
Comment (spetrunin) [inline apidef/api_definitions.go:655]: Please and this switch on a Supegraph config via simple bool var

File: apidef/api_definitions.go [hunk 2]
```diff
@@ -762,6 +764,12 @@ type GraphQLPlayground struct {
 	Path string `bson:"path" json:"path"`
 }
 
+// GraphQLQueryBatching is a configuration for query batching
+// https://www.apollographql.com/blog/apollo-client/performance/batching-client-graphql-queries/
```
Comment (spetrunin) [inline apidef/api_definitions.go:768]: please remove link to apollo

File: apidef/api_definitions.go [hunk 3]
```diff
@@ -671,10 +671,11 @@ type GraphQLSubgraphConfig struct {
...
 	// UpdatedAt contains the date and time of the last update of a supergraph API.
-	UpdatedAt     *time.Time              `bson:"updated_at" json:"updated_at,omitempty"`
-	Subgraphs     []GraphQLSubgraphEntity `bson:"subgraphs" json:"subgraphs"`
-	MergedSDL     string                  `bson:"merged_sdl" json:"merged_sdl"`
-	GlobalHeaders map[string]string       `bson:"global_headers" json:"global_headers"`
+	UpdatedAt             *time.Time              `bson:"updated_at" json:"updated_at,omitempty"`
+	Subgraphs             []GraphQLSubgraphEntity `bson:"subgraphs" json:"subgraphs"`
+	MergedSDL             string                  `bson:"merged_sdl" json:"merged_sdl"`
+	GlobalHeaders         map[string]string       `bson:"global_headers" json:"global_headers"`
+	DisableQueryBactching bool                    `bson:"disable_query_batching" json:"disable_query_batching"`
```
Comment (pvormste) [inline apidef/api_definitions.go:678]: Here's a typo: Should be `DisableQueryBatching`

**Review by pvormste — CHANGES_REQUESTED** [review]

**Review by pvormste — APPROVED** [review]

**Review by spetrunin — APPROVED** [review]

**Pull request merged**